src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/SharedPoolsOfWCFObjects.cs

```diff
@@ -0,0 +1,612 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.ServiceModel;
+using System.Text;
+using System.Threading;
+using System.Collections.Concurrent;
+
+namespace WCFClientStressTests
+{
+    #region Actual tests
+    // A full cycle of creating a pool of channel factories, using each factory to create
+    // a pool of channels, using all channels once and then closing all of them
+    public static class CreateAndCloseFactoryAndChannelFullCycleTest
+    {
+        private static CallStats s_CreateChannelStats = new CallStats();
+        private static CallStats s_CallChannelStats = new CallStats();
+        private static CallStats s_CloseChannelStats = new CallStats();
+        private static CallStats s_CloseFactoryStats = new CallStats();
+
+        static CreateAndCloseFactoryAndChannelFullCycleTest()
+        {
+        }
+        public static void CreateFactoriesAndChannelsUseAllOnceCloseAll()
+        {
+            using (var theOneTimeThing = new PoolOfThings<FactoryAndPoolOfItsObjects<ChannelFactory<WcfService1.IService1>, WcfService1.IService1>>(
+                    maxSize: 3, // # of pooled FactoryAndPoolOfItsObjects
```

Comment:
For future consideration -- make this "3" configurable or a static readonly int somewhere.  It's not clear to the reader why 3 is used.